### PR TITLE
🏗 Report brotli bundle-size

### LINKED ISSUE #23711: 🏗 Use Brotli-Size Directly
As we start to move from using `gzip` size to `brotli` size to determine the fluctuation a PR is introducing, we can lower our dependency count by switching to [brotli-size](https://github.com/erwinmombay/brotli-size#readme) directly.

Benefits:
1. Uses Node's builtin Brotli compression.
2. Reduces external dependency count.

### PR DESCRIPTION
**Pull request by danielrozenberg**
Progress on #21275 (see issue for work plan)
Fixes #23711

### CODE REVIEW

File: build-system/tasks/bundle-size.js [hunk 1]
```diff
@@ -45,17 +45,19 @@ const {red, cyan, yellow} = colors;
 /**
  * Get the gzipped bundle size of the current build.
```
Comment (rsimha) [inline build-system/tasks/bundle-size.js]: nit: Fix comment.

File: build-system/tasks/bundle-size.js [hunk 2]
```diff
@@ -188,7 +198,7 @@ async function skipBundleSize() {
 async function reportBundleSize() {
   if (isTravisPullRequestBuild()) {
     const baseSha = gitTravisMasterBaseline();
-    const bundleSize = parseFloat(getGzippedBundleSize());
+    const bundleSize = parseFloat(getBundleSize('gzip'));
```
Comment (rsimha) [inline build-system/tasks/bundle-size.js]: Worth adding a TODO here to change `gzip` to `brotli`?

Comment (danielrozenberg) [reply build-system/tasks/bundle-size.js]: Done.

File: build-system/tasks/bundle-size.js (hunk 1 above)
Comment (danielrozenberg) [reply build-system/tasks/bundle-size.js]: Done.

**Comment by rsimha** [comment]
You can avoid the CLA failures by making sure `git config user.email` is consistent across all your git clients.

**Comment by danielrozenberg** [comment]
I changed that config but the first commit in this PR is with my old email address 🤷‍♂

File: build-system/tasks/bundle-size.js [hunk 3]
```diff
@@ -99,53 +114,64 @@ function storeBundleSize() {
...
-              `ERROR: Failed to create the bundle-size/${commitHash} file in`
-            ),
-            red('the build artifacts repository on GitHub!')
-          );
-          log(red('Error message was:'), error.message);
-          process.exitCode = 1;
-        });
-    });
+        .catch(() => {
+          return octokit.repos
```
Comment (estherkim) [inline build-system/tasks/bundle-size.js]: nit: is it a quick change to unnest these by using await/catch? a bit easier to follow

File: build-system/tasks/bundle-size.js [hunk 4]
```diff
@@ -99,53 +114,64 @@ function storeBundleSize() {
...
-    .catch(() => {
-      return octokit.repos
-        .createOrUpdateFile(
-          Object.assign(githubApiCallOptions, {
-            message: `bundle-size: ${commitHash} (${bundleSize})`,
-            content: Buffer.from(bundleSize).toString('base64'),
-          })
-        )
+  const promises = [];
+  for (const [compression, extension, getBundleSize] of [
```
Comment (estherkim) [inline build-system/tasks/bundle-size.js]: 💁‍♂💁 I like this

Comment (danielrozenberg) [reply build-system/tasks/bundle-size.js]: Thank you? :P

File: build-system/tasks/bundle-size.js (hunk 3 above)
Comment (danielrozenberg) [reply build-system/tasks/bundle-size.js]: Done. PTAL!

File: build-system/tasks/bundle-size.js [hunk 5]
```diff
@@ -99,53 +114,64 @@ function storeBundleSize() {
...
-            red(
-              `ERROR: Failed to create the bundle-size/${commitHash} file in`
-            ),
-            red('the build artifacts repository on GitHub!')
-          );
-          log(red('Error message was:'), error.message);
-          process.exitCode = 1;
-        });
-    });
+        .catch(() => {
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply build-system/tasks/bundle-size.js]: 🤷‍♂

File: build-system/tasks/bundle-size.js (hunk 3 above)
Comment (estherkim) [reply build-system/tasks/bundle-size.js]: LG!

**Pull request merged**